app/lib/efile/md/md502b_calculator.rb

```diff
@@ -0,0 +1,34 @@
+module Efile
+  module Md
+    class Md502bCalculator < ::Efile::TaxCalculator
+      attr_reader :lines, :value_access_tracker
+
+      def initialize(value_access_tracker:, lines:, intake:)
+        @value_access_tracker = value_access_tracker
+        @lines = lines
+        @intake = intake
+      end
+
+      def calculate
+        set_line(:MD502B_LINE_1, :calculate_line_1)
+        set_line(:MD502B_LINE_2, :calculate_line_2)
+        set_line(:MD502B_LINE_3, :calculate_line_3)
+        @lines.transform_values(&:value)
```

Comment:
I'm still learning about calculators so this is more a question for my own edification. I thought these were only used for 'main' or 'top-level' calculators. When do we want to do this on a sub-calculator?

Reply:
oh good catch! this was a copy-paste mistake 🙈

Reply:
ok actually i added it back in because the main calculator pulls a line from this calculator (see `get_dependent_exemption_count` in md502_calculator). i remember when i added that i looked for an example of a calculator that references another calculator but i couldn't find one. wondering if there's a better way to do it...

Reply:
My current ticket is to do with tax calculators and I actually noticed how this gets worked around. I've got a little about it in my notes:

-   The `@d400_schedule_s` cvar must be calculated first to be able to be referenced for line 9
-   Note also that this class gets established in the `#initialize` method of the D400 calculator, passing its own cvar `@lines` in to be manipulated.
```ruby
    def initialize(year:, intake:, include_source: false)
      super
      @d400_schedule_s = Efile::Nc::D400ScheduleSCalculator.new(
        value_access_tracker: @value_access_tracker,
        lines: @lines,
        intake: @intake
      )
    end
```
-   In this way, the lines array is shared between the calculators and the &rsquo;wrapping&rsquo; calculator can freely access lines established by the &rsquo;internal&rsquo; calculator
-   Note that &rsquo;main&rsquo; forms will need to have some variant of `@lines.transform_values(&:value)` to return the values. Calculators used by the &rsquo;main&rsquo; calculator do not need to do this